### PR TITLE
Fix CSR validation failure for missing status conditions

### PR DESCRIPTION
**Pull request by mansikulkarni96**
This PR aims to fix the issue with failing CSR validation of CSR's that have a missing status condition.
 These CSR's cannot be approved, hence they need to be skipped.

Latest failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1258/pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-operator/1577062440414220288

csr.json: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_windows-machine-config-operator/1258/pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-operator/1577062440414220288/artifacts/vsphere-e2e-operator/gather-extra/artifacts/csr.json

"status": {}

### CODE REVIEW

File: test/e2e/validation_test.go [hunk 1]
```diff
@@ -665,6 +665,10 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 		return nil, errors.Wrap(err, "unable to get CSR list")
 	}
 	for _, c := range csrs.Items {
+		// skip the CSR if it's status has not been reported
+		if len(c.Status.Conditions) == 0 {
```
Comment (jrvaldes) [inline test/e2e/validation_test.go:669]: `Conditions` may be nil. Consider
```go
if c.Status.Conditions == nill || len(c.Status.Conditions) == 0 {
   ...
}
```

**Comment by mansikulkarni96** [comment]
> Thanks for working on this, LGTM so far. Will this be a temp workaround until we or the cloud team find a permanent fix?

That is correct @alinaryan

File: test/e2e/validation_test.go [hunk 2]
```diff
@@ -665,6 +665,10 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 		return nil, errors.Wrap(err, "unable to get CSR list")
 	}
 	for _, c := range csrs.Items {
+		// skip the CSR if it's status has not been reported
```
Comment (aravindhp) [inline test/e2e/validation_test.go:668]: Please add the suspected reason here and in the commit message.

**Review by aravindhp — APPROVED** [review]
Thanks for fixing this, @mansikulkarni96

**Comment by aravindhp** [comment]
Blocked by OCPBUGS-2175

**Comment by mtnbikenc** [comment]
@mansikulkarni96 the aws-e2e-operator job failed twice in the setup phase.  Is this a new issue with AWS or a known flake?

**Comment by mansikulkarni96** [comment]
There are issues creating EC2 instance, both jobs failed with different reasons though. Investigating.

**Comment by aravindhp** [comment]
/override ci/prow/aws-e2e-operator 

The test past previously we are now seeing a flake on AWS cluster installs.
```
level=error msg=Error: creating EC2 Instance: InvalidNetworkInterfaceID.NotFound
```

**Pull request merged**